src/BookInventory.java

```diff
@@ -0,0 +1,51 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class BookInventory {
+    public static void main(String[] args) {
+        List<Book> bookList = new ArrayList<>();
+        bookList.add(new Book("Pineapple Street", "Jenny Jackson", "red", 15.99));
+        bookList.add(new Book("Emily Henry", "Happy Place", "green", 12.50));
+        bookList.add(new Book("House of Earth and Blood", "Sarah J. Maas", "blue", 20.00));
+        bookList.add(new Book("If you could see the sun", "Ann Liang", "black", 8.75));
+        bookList.add(new Book("The midnight library", "Matt Haig", "red", 18.50));
+
+        Map<Integer, Book> bookMap = createBookMap(bookList);
+
+        System.out.println("All books:");
+        bookMap.values().forEach(System.out::println);
+
+        String targetColor = "red";
+        System.out.println("\nBooks with the color " + targetColor + ":");
+        List<Book> booksWithColor = getBooksByColor(bookMap, targetColor);
+        booksWithColor.forEach(System.out::println);
+
+        double targetPrice = 15.00;
+        System.out.println("\nBooks with the price more than " + targetPrice + ":");
+        List<Book> booksWithPriceAbove = getBooksByPriceAbove(bookMap, targetPrice);
+        booksWithPriceAbove.forEach(System.out::println);
+    }
+
+    public static Map<Integer, Book> createBookMap(List<Book> bookList) {
```

Comment:
ok but maybe this variant of implementation is optimal

Reply:
public static HashMap<Integer, String> createBookMap(ArrayList bookList) {
HashMap<Integer, String> bookMap = new HashMap<>();

    for (int i = 0; i < bookList.size(); i++) {
        bookMap.put(i, bookList.get(i));
    }
    return bookMap;
}